### PR TITLE
Bug fix/cookbook search.ap

### PR DESCRIPTION
**Pull request by Adelle-Pitsas**
What does this PR do?
- fixes bug that showed all recipe search results in cookbook search display
- adds clear button back in for more intuitive clear functionality

Where should your reviewer start?
- `searchReipeByName` on `line 295` through `line 313`

What (if any) features are you implementing?
- Adding a clear functionality: clear button is hidden on page load but appears when search has executed
- When clear button is clicked, all recipes or cookbook recipes appear with search button again.

What (if anything) did you refactor?

Were there any issues that arose?

Is there anything that you need from your teammate?

Any other comments, questions, or concerns to add?

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -287,16 +297,23 @@ function deselectTag() {
...
+    changeSearchButton(clearCookbookSearchButton, searchCookbookButton);
   }
 }
 
-function changeSearchButton() {
+function changeSearchButton(clearButton, searchButton) {
   cookbookSearchBar.value = '';
   allRecipesSearchBar.value = '';
+  show(clearButton)
+  hide(searchButton)
```
Comment (tricia-holmes) [inline src/scripts.js]: @Adelle-Pitsas I haven't been a part of the search so not sure how hard would this be! But could we make that it so it doesn't clear out on lines 313-314 `keyword` until you click the `clear` button? That way the user knows they are clearing out the search for that keyword. Right now it defaults to the original gray search message as soon as you search

**Review by tricia-holmes — CHANGES_REQUESTED** [review]

File: src/scripts.js (hunk 1 above)
Comment (Adelle-Pitsas) [reply src/scripts.js]: Yeah, can definitely do that!

Comment (Adelle-Pitsas) [reply src/scripts.js]: Updated so there is an event listener on the clear button to clear the search bar when clicked. Added a new helper function `clearSearchBar` on `line 326`

Comment (tricia-holmes) [reply src/scripts.js]: Thank you! Was this change in the PR? I pulled and checkout into the branch but don't see the clearSearchBar on line 326 and it clears the keyword on search. I may pulled down wrong on my end, but wanted to check!

Comment (Adelle-Pitsas) [reply src/scripts.js]: Just saw this-- you're right, I committed and didn't push up. Should be all set now!

Comment (tricia-holmes) [reply src/scripts.js]: Awesome! Thank you! Just one last thing - I think we need to add the `clearSearchBar` function for whenever the user clicks on the `All recipes tab` or the `cookbook tab`.  Right now if they don't follow our user flow of hitting the clear button, but instead click the tab then the keyword will be there when they go back. I'm pretty sure we just need to add it so when they click the tab it runs the clear search bar too. After that though, everything looks great! 

https://user-images.githubusercontent.com/108706408/197537843-d7593f9d-af30-410d-bba2-4f0891d64274.mp4

Comment (Adelle-Pitsas) [reply src/scripts.js]: Good catch! Just added it to the event listener function.

**Pull request merged**